src/QMCHamiltonians/NonLocalECPotential.h

```diff
@@ -2,37 +2,57 @@
 // This file is distributed under the University of Illinois/NCSA Open Source License.
 // See LICENSE file in top directory for details.
 //
-// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+// Copyright (c) 2022 QMCPACK developers.
 //
 // File developed by: Ken Esler, kpesler@gmail.com, University of Illinois at Urbana-Champaign
 //                    Jeremy McMinnis, jmcminis@gmail.com, University of Illinois at Urbana-Champaign
 //                    Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
 //                    Jaron T. Krogel, krogeljt@ornl.gov, Oak Ridge National Laboratory
 //                    Mark A. Berrill, berrillma@ornl.gov, Oak Ridge National Laboratory
+//                    Peter W. Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
 //
 // File created by: Jeongnim Kim, jeongnim.kim@gmail.com, University of Illinois at Urbana-Champaign
 //////////////////////////////////////////////////////////////////////////////////////
 
 
 #ifndef QMCPLUSPLUS_NONLOCAL_ECPOTENTIAL_H
 #define QMCPLUSPLUS_NONLOCAL_ECPOTENTIAL_H
+#include "Configuration.h"
 #include "QMCHamiltonians/NonLocalTOperator.h"
 #include "QMCHamiltonians/ForceBase.h"
 #include "QMCHamiltonians/NonLocalECPComponent.h"
 #include "Particle/NeighborLists.h"
-
 namespace qmcplusplus
 {
 template<typename T>
 struct NLPPJob;
 
-struct NonLocalECPotentialMultiWalkerResource;
+namespace testing
+{
+class TestNonLocalECPotential;
+}
 
 /** @ingroup hamiltonian
  * \brief Evaluate the semi local potentials
  */
 class NonLocalECPotential : public OperatorBase, public ForceBase
 {
+  using Real = QMCTraits::RealType;
+
+  struct NonLocalECPotentialMultiWalkerResource : public Resource
+  {
+    NonLocalECPotentialMultiWalkerResource() : Resource("NonLocalECPotential") {}
+
+    Resource* makeClone() const override;
+
+    ResourceCollection collection{"NLPPcollection"};
+
+    int num_walker = 0;
```

Comment:
Really need this? It is a high risk state. Can we keep the forward declaration and move the full definition back to cpp.

Reply:
Mistake on my part.  I've moved the definition back into the cpp, but it differs from the treatment of practically every other MultiWalkerResource.  Since it lets us make changes without causing as big a recompilation it is in principle good.